feat: use JSON Patch for flow updates

ogabrielluiz opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

## Release Notes

* **New Features**
  * Added JSON Patch support for flows, enabling efficient partial updates via a dedicated endpoint
  * Optimized flow save operations to reduce payload size by applying only changed fields
  * Enhanced flow update capabilities with RFC 6902 standard operations (add, remove, replace, move, copy, test)

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ogabrielluiz committed:
feat(flows): add JSON Patch support for flow updates

ogabrielluiz committed:
tests(flows): add JSON Patch endpoint tests

ogabrielluiz committed:
feat(flows): add JSON patch flow mutation for flow updates

ogabrielluiz committed:
feat(flows): add fast-json-patch dependency for flow updates

ogabrielluiz committed:
feat(flows): implement json patch hook

ogabrielluiz committed:
test(flows): add JSON patch utilities unit tests

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

ogabrielluiz committed:
perf(flows): optimize json-patch response with lightweight JsonPatchResponse

ogabrielluiz committed:
feat(flows): add JSON Patch types for RFC 6902 operations

ogabrielluiz committed:
feat(flows): add json patch endpoint for custom components

ogabrielluiz committed:
test(flows): add unit tests for custom component json patch endpoint

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

ogabrielluiz committed:
Revert "feat(flows): add json patch endpoint for custom components"

This reverts commit 2508fa6fbb99a4a508a0f434ef2ad8e52ff6bdf0.

ogabrielluiz committed:
feat(flows): add JsonPatchResponse schema for flow operations

ogabrielluiz committed:
refactor(patch-flows): remove json-patch utilities

jordanrfrazier committed:
Claude worked on the FE handling patch responses instead of full flow objects

ogabrielluiz committed:
refactor(flows): return patch operations instead of patched data in response

ogabrielluiz committed:
refactor(flows): apply patch operations client-side using fast-json-patch

ogabrielluiz committed:
refactor(flows): use jsondiffpatch with ID-aware array diffing

ogabrielluiz committed:
chore(frontend): add jsondiffpatch library for flow patch operations

ogabrielluiz committed:
test(flows): add JSON patch optimization spec for node deletion

Adam-Aghili commented on src/backend/base/langflow/api/v1/flows.py:
```diff
@@ -8,6 +8,7 @@
 from typing import Annotated
 from uuid import UUID
 
+import jsonpatch
```
Neither jsonpatch or anyio are inside of pyproject.toml and I don't understand how this isn't causing a build failure.

Adam-Aghili reviewed (changes requested):
I eventually get 400s and my changes don't save 

<video src="https://github.com/user-attachments/assets/6fec559d-6810-4e72-bbc3-0bd03ba9acb1" />

ogabrielluiz committed:
Address JSON Patch code review feedback

- Filter restricted fields (id, user_id, created_at, updated_at) instead of rejecting
- Add RFC 6902 value validation for add/replace/test operations
- Fix remove_api_keys call to work with correct flow data structure
- Add comprehensive unit tests for all JSON Patch operations
- Add e2e tests for node and edge deletion with minimal operation verification
- Fix frontend type import for JSON Patch operations

ogabrielluiz committed:
refactor: remove operations field from JsonPatchResponse and related tests

ogabrielluiz committed:
refactor(flows): enhance JSON patch handling with improved error reporting and field tracking

ogabrielluiz committed:
fix(flows): correct conditional check for folder refetching in usePatchJsonPatchFlow

ogabrielluiz committed:
refactor(tests): improve JSON Patch request/response handling and add failure tracking

ogabrielluiz committed:
chore: add .claude directory to .gitignore

ogabrielluiz committed:
chore: remove .claude/settings.local.json configuration file
